Source/DafnyLanguageServer/CounterExampleGeneration/DafnyModel.cs

```diff
@@ -124,23 +123,8 @@ public class DafnyModel {
     /// and collect all known datatype values
     /// </summary>
     private void InitArraysAndDataTypes() {
-      var arrayLength = new Regex("^_System.array[0-9]*.Length[0-9]*$");
       foreach (var fn in Model.Functions) {
-        if (arrayLength.IsMatch(fn.Name)) {
-          int j = fn.Name.IndexOf('.', 13);
-          int dims = j == 13 ? 1 : int.Parse(fn.Name.Substring(13, j - 13));
-          int idx = j == 13 ? 0 : int.Parse(fn.Name[(j + 7)..]);
-          foreach (var tpl in fn.Apps) {
-            var elt = tpl.Args[0];
-            var len = tpl.Result;
-            if (!arrayLengths.TryGetValue(elt, out var ar)) {
-              ar = new Model.Element[dims];
-              arrayLengths.Add(elt, ar);
-            }
-            Contract.Assert(ar[idx] == null);
-            ar[idx] = len;
-          }
-        } else if (fn.Name.StartsWith("#") && fn.Name.IndexOf('.') != -1 && fn.Name[1] != '#') {
+        if (fn.Name.StartsWith("#") && fn.Name.IndexOf('.') != -1 && fn.Name[1] != '#') {
```

Comment:
Could you put the magic string into named constants, so it's more clear where they come from?

Reply:
Done!